kcwitools/extract_optimal.py

```diff
@@ -0,0 +1,162 @@
+from __future__ import print_function
+
+import numpy as np
+from astropy.io import fits
+import matplotlib.pyplot as plt
+from kcwitools import optimal_extract_utility as r
+from linetools.spectra.xspectrum1d import XSpectrum1D
+
+
+def extract_optimal_1D(fluxcube, varcube,header,wavegrid,x1,x2,y1,y2,outfile=None,verbose=False,plot=False):
+	"""
+
+	Args:
+		fluxcube: 3D kcwi fluxcube
+		varcube: 3D KCWI variance cube
+		header:  header information
+		wavegrid:  1D kcwi wavelength grid
+		
+		x1,x2,y1,y2: Four corners of an aribaty rectangular pseudo-slit for extraction
+		outfile: str, optional
+		  FITS file name
+		verbose [False] : optional, prints information on terminal
+		plot [False] : optional plots diagnostic stuff
+	
+	Returns:
+		xspec: XSpectrum1D
+
+	"""	
+
+	# CHECK CHECK CHECK
+	#THESE VALUES NEED TO BE DOUBLE CHECKED. WHICH ONE SHOULD I USE?
+	gain=header['GAIN2']
+	rdnoise=3.5  #  Don't know this value. Please instruct what should I read in here?
+
+
+	# Hardcoding some values
+	sigma=5  # Sigma Clipping Value
+	fit_order=200   # Chebyshev polinomial order
+
+
+	# Create slices within wavelength ranges that are good for KCWI
+	# From now on only deal with these smaller slices
+	slices=np.where((wavegrid >=3500.) & (wavegrid <= 5500.))
+	flux=fluxcube[slices[0],:,:]
+	wave=wavegrid[slices[0]]
+	var_origin=(varcube[slices[0],:,:])
+
+
+
+	# Get the required shapes
+	wavedim, ydim, xdim = flux.shape
+
+
+	# Plot and show the slit position on the white light image
+	if plot==True:
+		whitelight=np.sum(flux, axis=0)
```

Comment:
Shouldn't we use our `whitelight` method instead?

Reply:
Done, and added     header['NAXIS3'] = int(sum(slices)), to reflect the new sliced wavegrid